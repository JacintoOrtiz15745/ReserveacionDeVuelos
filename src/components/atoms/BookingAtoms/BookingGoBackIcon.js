import React from 'react';
import {TouchableOpacity} from 'react-native';
import {colors} from '../../../styles/Colors';
import Icon from 'react-native-vector-icons/AntDesign';

const BookingGoBackIcon = ({navigation, RouteToGo}) => {
  return (
    <TouchableOpacity
      onPress={() => navigation.navigate(RouteToGo)}
      testID="GoBackBtn">
      <Icon name="left" color={colors.Purple} size={30} testID="GoBackIcon" />
    </TouchableOpacity>
  );
};

export default BookingGoBackIcon;
