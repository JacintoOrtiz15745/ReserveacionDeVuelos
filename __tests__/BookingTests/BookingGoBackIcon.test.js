import React from 'react';
import {shallow} from 'enzyme';
import BookingGoBackIcon from '../../src/components/atoms/BookingAtoms/BookingGoBackIcon';

describe('Component renders correctly', () => {
  const props = {
    navigation: {navigate: jest.fn()},
    RouteToGo: 'some route',
  };
  const wrapper = shallow(<BookingGoBackIcon {...props} />);
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('component exists', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'GoBackBtn',
    );
    expect(component.exists()).toBeTruthy();
  });
  test('navigate function gest called', () => {
    const component = wrapper.findWhere(
      (node) => node.prop('testID') === 'GoBackBtn',
    );
    component.props().onPress();
    wrapper.update();
    expect(props.navigation.navigate).toHaveBeenCalled();
  });
  test('icon exists', () => {
    const icon = wrapper.findWhere(
        (node) => node.prop('testID') === 'GoBackIcon',
      );
      expect(icon.exists()).toBeTruthy();
  });
});
