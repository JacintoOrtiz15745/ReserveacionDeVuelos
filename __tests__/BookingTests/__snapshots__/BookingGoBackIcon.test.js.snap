// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  testID="GoBackBtn"
>
  <Icon
    allowFontScaling={false}
    color="#6472ee"
    name="left"
    size={30}
    testID="GoBackIcon"
  />
</ForwardRef>
`;
